src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerManager.AnalyzerExecutionContext.cs

```diff
@@ -67,15 +71,19 @@ public Task<HostSessionStartAnalysisScope> GetSessionAnalysisScopeTask(AnalyzerE
                         return _lazySessionScopeTask;
                     }
 
-                    task = Task.Run(() =>
-                    {
-                        var sessionScope = new HostSessionStartAnalysisScope();
-                        analyzerExecutor.ExecuteInitializeMethod(_analyzer, sessionScope);
-                        return sessionScope;
-                    }, analyzerExecutor.CancellationToken);
-
+                    task = getSessionAnalysisScopeTaskSlow(this, analyzerExecutor);
```

Comment:
>task = getSessionAnalysisScopeTaskSlow(this, analyzerExecutor); [](start = 20, length = 63)

Would we achieve the same effect if we were to add a block around the original assignment and to add locals ```AnalyzerExecutionContext context, AnalyzerExecutor executor``` in that block?

Reply:
It would, but it's easy to accidentally miss the capture for `this` using that pattern. The static local function does a better job of providing visual assurance. See https://github.com/microsoft/perfview/pull/1015/files#r336253660 for example.